pxr/usdImaging/usdImaging/delegate.cpp

```diff
@@ -1082,6 +1082,39 @@ UsdImagingDelegate::ApplyPendingUpdates()
     indexProxy._UniqueifyPathsToRepopulate();
     _Populate(&indexProxy);
     _ExecuteWorkForVariabilityUpdate(&worker);
+
+    // Mark all dirty collection prims
+    if (_collectionCache.AreAllPathsDirty())
+    {
+        TRACE_SCOPE("Mark all dirty collection members");
+        TF_FOR_ALL(it, _hdPrimInfoMap) {
+            UsdImagingPrimAdapterSharedPtr& adapter = it->second.adapter;
+            TF_DEBUG(USDIMAGING_CHANGES).Msg("[Update]: invalidate collection member prim (all paths dirty) %s\n", it->first.GetText());
+            adapter->MarkDirty(it->second.usdPrim, it->first,
```

Comment:
We should only do this for gprims.  The standard way to do this is to add a "MarkCollectionsDirty" on prim adapter that does nothing, and an override on gprim adapter that calls MarkDirty(DirtyCategories).

"DirtyCollections" doesn't mean anything on other types of prim and will invalidate a random attribute.